From 12fe6a06c4c77e5c549444543aa3d4183f593628 Mon Sep 17 00:00:00 2001
From: Jeremie Graulle <jeremiex.graulle@intel.com>
Date: Tue, 18 Jun 2013 14:06:33 +0200
Subject: Add Android.mk command to import headers without import symbols

LOCAL_COPY_HEADERS is deprecated by Android, we have to use
LOCAL_EXPORT_C_INCLUDE_DIRS instead.
In test we need to import a header without link with the library.
We replace the body of the function by a stub or a mock.
But we need to have the same prototype of the function by importing the
header. But there are no way to import a header exported by a
LOCAL_EXPORT_C_INCLUDE_DIRS without link with the library.

This patch add 2 new LOCAL variables
LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES and
LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES to import include
from libraries without import symbols.

Bug: 117000
Domain: AOSP-build
Category: AOSP Improvement
Source: Internal
Upstream-candidate: No

Change-Id: I14355dd74a66c99b361335b7044becc1f8d1fb35
Original-Change-Id: I6f668c900003e27ba3c2ce0d2ab9200b6f058faf
Signed-off-by: Jeremie Graulle <jeremiex.graulle@intel.com>
Reviewed-on: http://android.intel.com:8080/114072
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Rao, Chetan <chetan.rao@intel.com>
Reviewed-by: De Chivre, Renaud <renaud.de.chivre@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 core/binary.mk     |    6 +++++-
 core/clear_vars.mk |    2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/binary.mk b/core/binary.mk
index d339317..6f98243 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -783,7 +783,11 @@ import_includes_deps := $(strip \
     $(foreach l, $(installed_shared_library_module_names), \
       $(call intermediates-dir-for,SHARED_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE),,$(LOCAL_2ND_ARCH_VAR_PREFIX))/export_includes) \
     $(foreach l, $(my_static_libraries) $(my_whole_static_libraries), \
-      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE),,$(LOCAL_2ND_ARCH_VAR_PREFIX))/export_includes))
+      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE),,$(LOCAL_2ND_ARCH_VAR_PREFIX))/export_includes) \
+    $(foreach l, $(LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES), \
+      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE),,$(LOCAL_2ND_ARCH_VAR_PREFIX))/export_includes) \
+    $(foreach l, $(LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES), \
+      $(call intermediates-dir-for,SHARED_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE),,$(LOCAL_2ND_ARCH_VAR_PREFIX))/export_includes))
 $(import_includes) : $(import_includes_deps)
 	@echo Import includes file: $@
 	$(hide) mkdir -p $(dir $@) && rm -f $@
diff --git a/core/clear_vars.mk b/core/clear_vars.mk
index f23c4a6..fa2b6fa 100644
--- a/core/clear_vars.mk
+++ b/core/clear_vars.mk
@@ -50,6 +50,8 @@ LOCAL_CONLYFLAGS:=
 LOCAL_RTTI_FLAG:=
 LOCAL_C_INCLUDES:=
 LOCAL_EXPORT_C_INCLUDE_DIRS:=
+LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES :=
+LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES :=
 LOCAL_LDFLAGS:=
 LOCAL_LDLIBS:=
 LOCAL_AAPT_FLAGS:=
-- 
1.7.10.4

